src/logic/calculate.js

```diff
@@ -0,0 +1,27 @@
+import operate from './operate';
+
+const calculate = (data, buttonName) => {
+  const { total, next, operation } = data;
+
+  switch (buttonName) {
+    case '+/-':
+      return operate(total, next, operation);
+
+    case '+':
+      return operate(total, next, operation);
+
+    case '-':
+      return operate(total, next, operation);
+
+    case '÷':
+      return operate(total, next, operation);
+
+    case '%':
+      return operate(total, next, operation);
+
+    default:
+      return 'ERROR';
+  }
+};
+
+export default calculate;
```

Comment:
- [ ] The calculate function is missing some of its logic. Right now is not evaluating the `buttonName`. Here we need to cover all cases for all button names (AC, =, dot button). And call `operate` only when the an operation needs to be done.

Reply:
Thanks for the review, @Torres-ssf. Changes have been effected.